### PR TITLE
add support for MOQO-based ZEAG Energie AG

### PR DESCRIPTION
**Pull request by ThorstenFroehlinghaus**
new MOQO feed

### CODE REVIEW

File: x2gbfs/providers/moqo.py [hunk 1]
```diff
@@ -295,3 +301,7 @@ def _pricing_plan_ids(self, vehicle_type: str) -> list[str]:
 
 class StadtwerkTauberfrankenProvider(MoqoProvider):
     pass
+
+
+class ZeagEnergieProvider(MoqoProvider):
+    pass
```
Comment (hbruch) [inline x2gbfs/providers/moqo.py:307]: As long as the superclass' default behavior is sufficient, you even would not need to define a subclass. For now, we could get rid of `StadtwerkTauberfrankenProvider`and `ZeagEnergieProvider` and just use `MoqoProvider`

File: x2gbfs/x2gbfs.py [hunk 2]
```diff
@@ -20,6 +20,7 @@
     NoiProvider,
     StadtwerkTauberfrankenProvider,
     VoiRaumobil,
+    ZeagEnergieProvider,
```
Comment (hbruch) [inline x2gbfs/x2gbfs.py:23]: I suggest to switch to `MoqoProvider`

File: x2gbfs/x2gbfs.py [hunk 3]
```diff
@@ -49,6 +50,8 @@ def build_extractor(provider: str, feed_config: Dict[str, Any]) -> BaseProvider:
         return CantamenIXSIProvider(feed_config)
     if provider in ['stadtwerk_tauberfranken']:
         return StadtwerkTauberfrankenProvider(feed_config)
+    if provider in ['zeag_energie']:
+        return ZeagEnergieProvider(feed_config)
```
Comment (hbruch) [inline x2gbfs/x2gbfs.py:54]: Here as well

**Review by hbruch — CHANGES_REQUESTED** [review]
I suggest to get rid of the currently needless provider specific subclasses.

And depending on the expected address properties, it might be appropriate to be a bit more defensive in accessing them.

Be aware, that the GBFS spec defines `address` as:

> 
Address (street number and name) where station is located. This MUST be a valid address, not a free-form text description. Example: 1234 Main Street

While currently no `city`, but just a `post_code` attribute is defined, I added only the city to the address, and provided the postal code via `post_code`.

**Comment by ThorstenFroehlinghaus** [comment]
Ok, I removed the subclasses and changed the address format.

**Review by hbruch — APPROVED** [review]
LGTM

**Pull request merged**